panel/src/helpers/upload.js

```diff
@@ -1,3 +1,45 @@
+import { random } from "./string.js";
+
+/**
+ * Uploads a file in chunks
+ * @param {File} file - file to upload
+ * @param {Object} params - upload options (see `upload` method for details)
+ * @param {number} size - chunk size in bytes
+ */
+export async function chunked(file, params, size = 5242880) {
+	const parts = Math.ceil(file.size / size);
+	const id = random(4).toLowerCase();
+	let response;
+
+	for (let i = 0; i < parts; i++) {
+		const start = i * size;
+		const end = Math.min(start + size, file.size);
+		const chunk = file.slice(start, end, file.type);
```

Comment:
How does this slicing work? Does the browser stream the file to the server or does it need to read the chunk into client memory?

Reply:
Using https://developer.mozilla.org/en-US/docs/Web/API/Blob/slice

Reply:
OK, so that reads the chunk into client memory. There seems to be a streaming API for files and for `fetch`, but in a quick search I couldn't find a example to plug them together.

Maybe we should limit the chunk size to something like 5 MB in the meantime in the `Upload::chunkSize()` method to avoid overloading the browser. Especially mobile devices may not always have enough memory to handle 100 MB or even larger requests.

Reply:
I've looks around and all implementations of chunks uploads, incl. tus (https://github.com/tus/tus-js-client/blob/main/lib/browser/sources/FileSource.js#L19), are using it the same way. I don't think we need to worry here and shouldn't overengineer it.

Reply:
That's great then!